airflow/models/abstractoperator.py

```diff
@@ -730,12 +733,15 @@ def _do_render_template_fields(
                 pass
 
             try:
-                rendered_content = self.render_template(
-                    value,
-                    context,
-                    jinja_env,
-                    seen_oids,
-                )
+                if callable(value):
+                    rendered_content = value(self.render_value, context, jinja_env)
```

Comment:
I think this is enough:

```suggestion
                    rendered_content = value(context=context, jinja_env=jinja_env)
```

The current operator is available in `context["task"]` so we can instruct users to call `render_value` on that with the appropriate arguments if they need to render templates inside the callable.

Also it might be a good idea to always pass named arguments (and in the documentation tell the users to always use these names) so it’s easier to extend the interface in the future if we need to.